### PR TITLE
squid: doc/cephadm: add default monitor images

### PR DESCRIPTION
**Pull request by zdover23**
Add a list of default monitor images to the documentation. This commit is made in response to a request from Eugen Block, and is made using the information developed by Mr Block here:
https://lists.ceph.io/hyperkitty/list/ceph-users@ceph.io/thread/QGC66QIFBKRTPZAQMQEYFXOGZJ7RLWBN/.

Signed-off-by: Zac Dover <zac.dover@proton.me>
(cherry picked from commit d6311d24c277f35fcae5097ed8a19843e62e6149)

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by anthonyeleven — APPROVED** [review]

File: doc/cephadm/services/monitoring.rst
Comment (adk3798) [inline doc/cephadm/services/monitoring.rst:176]: I think this only works in pacific and quincy, where the cephadm isn't a zip file. In main/squid, I found this process works instead
```
root@smithi027:~# mkdir cephadm_dir
root@smithi027:~# 
root@smithi027:~# unzip /var/lib/ceph/b9943a26-07e0-11ef-bc96-c7b262605968/cephadm.8a4210adf49bf7cbbb308ea609b0f309661d65fe037af8a6095519adb47c62fe -d cephadm_dir > /dev/ull
warning [/var/lib/ceph/b9943a26-07e0-11ef-bc96-c7b262605968/cephadm.8a4210adf49bf7cbbb308ea609b0f309661d65fe037af8a6095519adb47c62fe]:  14 extra bytes at beginning or within zipfile
  (attempting to process anyway)
root@smithi027:~# 
root@smithi027:~# egrep "_IMAGE" cephadm_dir/cephadmlib/constants.py
DEFAULT_IMAGE = 'quay.ceph.io/ceph-ci/ceph:main'
DEFAULT_IMAGE_IS_MAIN = True
DEFAULT_IMAGE_RELEASE = 'squid'
DEFAULT_PROMETHEUS_IMAGE = 'quay.io/prometheus/prometheus:v2.43.0'
DEFAULT_LOKI_IMAGE = 'docker.io/grafana/loki:2.4.0'
DEFAULT_PROMTAIL_IMAGE = 'docker.io/grafana/promtail:2.4.0'
DEFAULT_NODE_EXPORTER_IMAGE = 'quay.io/prometheus/node-exporter:v1.5.0'
DEFAULT_ALERT_MANAGER_IMAGE = 'quay.io/prometheus/alertmanager:v0.25.0'
DEFAULT_GRAFANA_IMAGE = 'quay.io/ceph/grafana:9.4.12'
DEFAULT_HAPROXY_IMAGE = 'quay.io/ceph/haproxy:2.3'
DEFAULT_KEEPALIVED_IMAGE = 'quay.io/ceph/keepalived:2.2.4'
DEFAULT_NVMEOF_IMAGE = 'quay.io/ceph/nvmeof:1.2.1'
DEFAULT_SNMP_GATEWAY_IMAGE = 'docker.io/maxwo/snmp-notifier:v1.2.1'
DEFAULT_ELASTICSEARCH_IMAGE = 'quay.io/omrizeneva/elasticsearch:6.8.23'
DEFAULT_JAEGER_COLLECTOR_IMAGE = 'quay.io/jaegertracing/jaeger-collector:1.29'
DEFAULT_JAEGER_AGENT_IMAGE = 'quay.io/jaegertracing/jaeger-agent:1.29'
DEFAULT_JAEGER_QUERY_IMAGE = 'quay.io/jaegertracing/jaeger-query:1.29'
DEFAULT_SMB_IMAGE = 'quay.io/samba.org/samba-server:devbuilds-centos-amd64'
```

on reef the process is similar, although you would run the grep against `cephadm_dir/__main__.py` rather than `cephadm_dir/cephadmlib/constants.py`

On pacific and quincy, the procedure you have here I think works, although I couldn't get this specific `grep -E DEFAULT*IMAGE` to work for me. I had better luck with `grep -E "_IMAGE"

```
root@smithi027:~# grep -E "DEFAULT*IMAGE" cephadm_dir/cephadmlib/constants.py
root@smithi027:~# 
root@smithi027:~# grep -E "_IMAGE" cephadm_dir/cephadmlib/constants.py
DEFAULT_IMAGE = 'quay.ceph.io/ceph-ci/ceph:main'
DEFAULT_IMAGE_IS_MAIN = True
DEFAULT_IMAGE_RELEASE = 'squid'
DEFAULT_PROMETHEUS_IMAGE = 'quay.io/prometheus/prometheus:v2.43.0'
DEFAULT_LOKI_IMAGE = 'docker.io/grafana/loki:2.4.0'
DEFAULT_PROMTAIL_IMAGE = 'docker.io/grafana/promtail:2.4.0'
DEFAULT_NODE_EXPORTER_IMAGE = 'quay.io/prometheus/node-exporter:v1.5.0'
DEFAULT_ALERT_MANAGER_IMAGE = 'quay.io/prometheus/alertmanager:v0.25.0'
DEFAULT_GRAFANA_IMAGE = 'quay.io/ceph/grafana:9.4.12'
DEFAULT_HAPROXY_IMAGE = 'quay.io/ceph/haproxy:2.3'
DEFAULT_KEEPALIVED_IMAGE = 'quay.io/ceph/keepalived:2.2.4'
DEFAULT_NVMEOF_IMAGE = 'quay.io/ceph/nvmeof:1.2.1'
DEFAULT_SNMP_GATEWAY_IMAGE = 'docker.io/maxwo/snmp-notifier:v1.2.1'
DEFAULT_ELASTICSEARCH_IMAGE = 'quay.io/omrizeneva/elasticsearch:6.8.23'
DEFAULT_JAEGER_COLLECTOR_IMAGE = 'quay.io/jaegertracing/jaeger-collector:1.29'
DEFAULT_JAEGER_AGENT_IMAGE = 'quay.io/jaegertracing/jaeger-agent:1.29'
DEFAULT_JAEGER_QUERY_IMAGE = 'quay.io/jaegertracing/jaeger-query:1.29'
DEFAULT_SMB_IMAGE = 'quay.io/samba.org/samba-server:devbuilds-centos-amd64'
```

**Review by adk3798 — CHANGES_REQUESTED** [review]

File: doc/cephadm/services/monitoring.rst
Comment (adk3798) [reply doc/cephadm/services/monitoring.rst:176]: Just remembered this is a backport. We'd need to make a change in main as well to account for this. Do we want to merge this as is and then apply a patch to main and bring it down, or make an adjustment here before merging @zdover23 ? If the former, feel free to dismiss my review so you can just merge this.

Comment (zdover23) [reply doc/cephadm/services/monitoring.rst:176]: Thanks for this @adk3798. I'll make these alterations (and add material to all the current release branches to explain these differences) later in the day.

Comment (zdover23) [reply doc/cephadm/services/monitoring.rst:176]: > Just remembered this is a backport. We'd need to make a change in main as well to account for this. Do we want to merge this as is and then apply a patch to main and bring it down, or make an adjustment here before merging @zdover23 ? If the former, feel free to dismiss my review so you can just merge this.

I'll merge it and then make the alterations you suggested. The docs won't display this inaccuracy for more than twelve hours. Fortunately, the docs aren't as hard-nosed as the code, so they can afford to be a palimpsest.

Comment (adk3798) [reply doc/cephadm/services/monitoring.rst:176]: Works for me. Dismissed by review

Comment (zdover23) [reply doc/cephadm/services/monitoring.rst:176]: https://github.com/ceph/ceph/pull/57225 - Squid procedure

Comment (zdover23) [reply doc/cephadm/services/monitoring.rst:176]: https://github.com/ceph/ceph/pull/57236 - Reef procedure

Comment (phlogistonjohn) [reply doc/cephadm/services/monitoring.rst:176]: Jostling my way into this nearly cold discussion:  Adam's procedure will work but I don't think the dev team is going to guarantee that these constants will exist in that location indefinitely. 

Personally, I would like to see the code make it simpler to accomplish the task of listing images (default and current) and making it easier to mirror images... but that's a task for the future.

Comment (zdover23) [reply doc/cephadm/services/monitoring.rst:176]: @phlogistonjohn, I'll keep abreast of this and make sure that the docs are as current as the shifting sands of technology permit.

**Pull request merged**
